### ISSUE TITLE
Wrong css links

### ISSUE THREAD

**Issue by kfo1182** [Postleaf/postleaf#81]
### Summary

CSS links in a fresh install don't include the port number. To make them work, one should either edit the code by hand, or start postleaf on the port 80 - with sudo.

### Steps to Reproduce

1. Install postleaf according to instruction.
2. Set APP_URL=http://localhost APP_PORT=3000 APP_HOST=localhost
 in the .env file.
3. Start postleaf and go to http://localhost:3000/admin
4. Manually replacing ```href="{@url...}"``` in views with ```href="/..."``` resolves the problem. 

### Additional info

- Postleaf version: master branch as of 28 march 2018
- Node version: 9.5.0
- Affected browsers: All of them
- Operating system: Gallium OS 4  (x86_64)

---

Note: This issue tracker is ONLY for bug reports and feature requests. If this is a personal support issue, please visit postleaf.org/support.

**Comment by claviska** [comment]
If you’re running Postleaf locally, the solution is to include the port
number in APP_URL in your .env file:

APP_URL=http://localhost:3000/

This only affects local installs, as most production installs are proxied
behind Nginx or Apache.

**Comment by kfo1182** [comment]
I certainly do the same thing on the production server. However, if there is an .env variable - why not use it? It wouldn't break reverse proxying, since nginx internally adresses node app with a port specified, but will fix the dev-server deployment. Current version of ```.env``` seems to be confusing.

**Comment by claviska** [comment]
It’s because APP_URL is strictly used to generate URLs.

Postleaf has no way of knowing whether not the port should be included in
the URL (not all dev environments will require it.)

This has been a point of confusion for some people but I’m not sure what
the best solution is. Open to ideas!

**Comment by kfo1182** [comment]
For any practical reason, do you have to know APP_URL to generate ```href```s for stattic assets? Do I miss something, or could you refactor ```.env``` variables in the following way: ```APP_URL``` dissapears, since you can always link static assets relatively to the host or ask your node.js instance what the host is.
If there is a need to run postleaf in a domain subfolder, you could create an ```APP_DOMAIN_SUBFOLDER``` var (i.e. ```=/blog/```), and concatenate it with relative urls.
```APP_HOST``` and ```APP_PORT``` stay what they are - settings to start up a node.js server.

Whatever the ```.env``` format gonna be, it may be helpful to leave some comments in ```.env.example```.

Excuse me if overlooked something obvious.

**Comment by claviska** [comment]
There are a handful of reasons we need to know the host (password reset emails, for example).

In the PHP version of Postleaf (now forked and called Leafpub), [we moved to schemaless URLs](https://github.com/Leafpub/leafpub/issues/57) but that change [was reverted](https://github.com/Leafpub/leafpub/issues/78) for the same reason.

Again, maybe `.env` isn't the best place for it. Maybe the proxy should be required to pass the hostname to the app (but then you'd still need to make it available in dev somehow). This approach just seemed practical at the time.
